### PR TITLE
Tyr Update Again

### PR DESCRIPTION
**Pull request by FluffMedic**
Aaaaaa
## About The Pull Request
A long list of changes, might be more if energy comes to me.
--Mecha attack patterns have been altered, uses call backs.
--Two of them have loot piles.
--Overseer has a better sprite.
--HoS replaces Overseer's old spot
--Medical temp removed
--Framework of new critters added (This and the previous is more because of lack of energy to finish lol)
--Overseer has a new attack pattern and is phase 4 of the Janus mecha.
--The right side has been expanded.
--More decals have been added.
--Beetle stunlock removed
--The rainbow firefly has it's terror radius increased.

## Changelog
:cl:
add: More things to Tyr
/:cl:

### CODE REVIEW

File: modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/mechas.dm [hunk 1]
```diff
@@ -317,162 +330,205 @@
 	downfour(target)
 	rightfour(target)
 	leftfour(target)
-	sleep(15)
 
 /mob/living/simple_mob/mechanical/mecha/eclipse/antipersonal_unit/proc/phase_two(atom/target) //SPIIIIN TO WIN
+	visible_message(span_danger("The [src] is preparing a deadly attack!"))
+	addtimer(CALLBACK(src, PROC_REF(spin_to_win), target), 2 SECOND, TIMER_DELETE_ME)
```
Comment (Kashargul) [inline modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/mechas.dm:336]: I know it's just semantic, but 0.5, 2, etc should use SECONDS as plural form.

Comment (FluffMedic) [reply modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/mechas.dm:336]: Alright will alter in a moment

Comment (Kashargul) [reply modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/mechas.dm:336]: thanks^^

File: modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/heads.dm [hunk 2]
```diff
@@ -531,31 +541,142 @@
...
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "drop_marker"
+
+/obj/effect/artillery_attack/Initialize(mapload)
+	..()
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/effect/artillery_attack/LateInitialize()
+	var/delay = rand(25, 30)
+	spawn(delay-7)
```
Comment (Kashargul) [inline modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/heads.dm:658]: could also be callbacked

I also wouldn't subtract and rather add on the deletion proc, it gives more freedom to play with numbers as you can't go negative

File: modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/mechas.dm [hunk 3]
```diff
@@ -506,523 +571,260 @@
...
+
+/mob/living/simple_mob/mechanical/mecha/eclipse/proc/beglaser_13(atom/target, var/next_cycle)
 	upfour_leftfour(target)
 	upfour_leftthree(target)
-	upfour_leftthree(target)
-	attackcycle = 3
+	upfour_lefttwo(target)
+	if(prob(50))
+		addtimer(CALLBACK(src, PROC_REF(beglaser), target, next_cycle), 1 SECOND, TIMER_DELETE_ME)
+	else
```
Comment (Kashargul) [inline modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/mechas.dm:755]: empty else?

Comment (FluffMedic) [reply modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/mechas.dm:755]: ac k, I am dumb. Thank you for catching that.

File: modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/eclipse.dm [hunk 4]
```diff
@@ -876,6 +876,7 @@
 	special_attack_max_range = 7
 
 /mob/living/simple_mob/humanoid/eclipse/solar/plant/do_special_attack(atom/A)
+	var/mob/living/L = A
 	visible_message(span_warning("\The [src]'s vines spread out!"))
 	Beam(A, icon_state = "vine", time = 3 SECONDS, maxdistance = INFINITY)
 	if(isliving(A))
```
Comment (Kashargul) [inline modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/eclipse.dm:882]: iscarbon right here. No need to check both.

iscarbon(A)

File: modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/eclipse.dm [hunk 5]
```diff
@@ -876,6 +876,7 @@
 	special_attack_max_range = 7
 
 /mob/living/simple_mob/humanoid/eclipse/solar/plant/do_special_attack(atom/A)
+	var/mob/living/L = A
```
Comment (Kashargul) [inline modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/eclipse.dm:879]: Not needed.

Comment (FluffMedic) [reply modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/eclipse.dm]: Fixed

Comment (FluffMedic) [reply modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/eclipse.dm]: Nevermind missed something

File: modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/eclipse.dm [hunk 6]
```diff
@@ -873,6 +876,14 @@
 	special_attack_max_range = 7
 
 /mob/living/simple_mob/humanoid/eclipse/solar/plant/do_special_attack(atom/A)
+	visible_message(span_warning("\The [src]'s vines spread out!"))
+	Beam(A, icon_state = "vine", time = 3 SECONDS, maxdistance = INFINITY)
+	if(isliving(A))
+		if(iscarbon(A))
+			addtimer(CALLBACK(src, PROC_REF(itemyoink), A), 3 SECOND, TIMER_DELETE_ME)
```
Comment (Kashargul) [inline modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/eclipse.dm:883]: SECONDS

File: modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/eclipse.dm [hunk 7]
```diff
@@ -953,9 +965,8 @@
 /mob/living/simple_mob/humanoid/eclipse/solar/nuclear/do_special_attack(atom/A)
 	visible_message(span_warning("\The [src] begins to glow green!"))
 	Beam(A, icon_state = "sat_beam", time = 3 SECONDS, maxdistance = INFINITY)
-	sleep(30)
-	var/obj/item/projectile/P = new /obj/item/projectile/beam/energy_net(get_turf(src))
-	P.launch_projectile(A, BP_TORSO, src)
+	addtimer(CALLBACK(src, PROC_REF(special_projectile), A), 3 SECOND, TIMER_DELETE_ME)
```
Comment (Kashargul) [inline modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/eclipse.dm:968]: SECONDS

File: modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/eclipse.dm [hunk 8]
```diff
@@ -1027,19 +998,25 @@
...
 	visible_message(span_danger("\The [src] begins to mess with a wrist mounted device."))
-	sleep(30)
-	if(isliving(A))
-		if(iscarbon(L))
-			return
-		else if(issilicon(L))
-			if(isrobot(L))
-				L.Weaken(10)
+	if(isrobot(A))
+		addtimer(CALLBACK(src, PROC_REF(remote_shutdown), A), 3 SECOND, TIMER_DELETE_ME)
```
Comment (Kashargul) [inline modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/eclipse.dm:1007]: SECONDS

**Comment by FluffMedic** [comment]
Sorry about the seconds thing.

**Comment by Kashargul** [comment]
> Sorry about the seconds thing.

No worries, it's minor, just easier for me to search when something should be longer than 1 second or shorter

**Comment by Kashargul** [comment]
one thing that should be considered, procs after a callback with a given object/mob should have a quick sanity check. 

so target is any mob, object, etc we gave go the timer. Then we return right away once the timer is triggered.
if(!target) return

as we have a timer, the target might (even if very rarely) get deleted before the timer proc fires

**Pull request merged**